### ISSUE TITLE
[JSR223] Saving a previously loaded rule file duplicates rules

### ISSUE THREAD

**Issue by openhab-5iver** [eclipse/smarthome#5981]
OH snapshot 1321 (ESH 0.10.0.201807241640)

The log appear to show that rules are removed and readded successfully...
```
2018-07-29 13:00:12.068 [INFO ] [cript.rulesupport.shared.ScriptedAutomationManager] - removeAll added handlers
2018-07-29 13:00:12.071 [INFO ] [ript.rulesupport.internal.loader.ScriptFileWatcher] - Loading script 'test.py'
2018-07-29 13:00:12.181 [DEBUG] [org.eclipse.smarthome.automation.module.core      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.events.EventSubscriber}={service.id=883, service.bundleid=212, service.scope=singleton, event.topics=smarthome/items/*} - org.eclipse.smarthome.automation.module.core
```
But triggering a rule shows that it is executed twice...
```
2018-07-29 13:00:16.609 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName1' of rule 'ac960942-ee26-4af8-888d-c407c72302c7' is triggered.
2018-07-29 13:00:16.611 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName1' of rule '798efe0a-e626-466c-b187-e6f94f720bfc' is triggered.
```
Another instance appears every time the rule file is saved. This possibly contributes to what was [reported here](https://github.com/eclipse/smarthome/issues/5939#issuecomment-408590292).

**Comment by openhab-5iver** [comment]
In looking at this further, rules are not running twice, so maybe it is just the trigger that is not being removed properly? Here is the log after removing all but one rule with one trigger. The current and old triggers fire, but only the existing rule is executed.
```
2018-07-29 16:37:35.296 [DEBUG] [ript.rulesupport.internal.loader.ScriptFileWatcher] - Script loaded: test.py
2018-07-29 16:37:46.330 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName2' of rule '8bcb13fc-de8c-461a-a5a4-ed6e45640ede' is triggered.
2018-07-29 16:37:46.330 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName2' of rule '11d30fb7-2920-4d41-9d29-defd3ae27b46' is triggered.
2018-07-29 16:37:46.330 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName1' of rule '8849c300-f07c-4531-bff8-a3c151f2886e' is triggered.
2018-07-29 16:37:46.330 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName1' of rule 'c4fb546c-82f0-4f67-8f2c-5fcc43164b87' is triggered.
2018-07-29 16:37:46.331 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The trigger 'testTriggerName2' of rule '977df89a-b612-40ba-8685-0e25e25bac55' is triggered.
2018-07-29 16:37:46.333 [DEBUG] [.smarthome.automation.core.internal.RuleEngineImpl] - The rule '977df89a-b612-40ba-8685-0e25e25bac55' is executed.
```

**Comment by openhab-5iver** [comment]
This appears to be resolved in OH snapshot 1325 (ESH 0.10.0.201808011124).
